storage/include/sparse_merkle/base_types.h

```diff
@@ -43,16 +55,27 @@ class Nibble {
     data_ = byte;
   }
 
+  // Get the bit of the Nibble starting from LSB.
+  // Bits 0-3 are available.
   bool getBit(const size_t bit) const {
     Assert(bit < SIZE_IN_BITS);
     return (data_ >> bit) & 1;
   }
 
   bool operator==(const Nibble& other) const { return data_ == other.data_; }
+  bool operator<(const Nibble& other) const { return data_ < other.data_; }
 
   // Return the underlying representation
   uint8_t data() const { return data_; }
 
+  // Return the nibble as its lowercase hex character.
+  char hexChar() const {
+    if (data_ >= 0 && data_ <= 9) {
+      return '0' + data_;
+    }
+    return 'a' + (data_ - 10);
```

Comment:
Do we want to verify that data_ does not exceed 0x0F?

Reply:
It's already a constraint on construction of the Nibble.